dali/kernels/imgproc/convolution/convolution_cpu.h

```diff
@@ -185,7 +185,7 @@ void ConvolveInnerDim(Out* out, const In* in, const W* window, int window_size,
     // Left border
     for (; x0 < 0 && xout < axis_size; x0++, xout++) {
       for (int c = 0; c < channels; c++) {
-        float acc = 0;
+        W acc = 0;
```

Comment:
Why? It's dangerous! What if the window is, say, uint8? What if the input is double and window is float? You'll lose precision or even overflow.

Reply:
Will revert, but I wonder, why it isn't something like `decltype(In{} * W{})` in the first place. (Not that it works well with the fabulous float16).

Reply:
Even with `decltype` we might end up with an overflow - consider uint16. The result would be int (or unsigned?) and you can still overflow when adding.

Reply:
Done.